### PR TITLE
Add mypy to CI

### PR DESCRIPTION
**Pull request by lightme16**

### CODE REVIEW

File: tg/main.py [hunk 1]
```diff
@@ -2,10 +2,10 @@
 import logging.handlers
 import signal
 import threading
-from curses import window, wrapper
+from curses import window, wrapper  # type: ignore
 from functools import partial
 
-from telegram.client import Telegram
+from telegram.client import Telegram  # type: ignore
```
Comment (paul-nameless) [inline tg/main.py:8]: what kind of error these ignoring?
Maybe we can add it to mypy config?
Also, let's disable check for long lines, it's annoying. Black fixes them if it's possible.

File: tg/views/__init__.py [hunk 2]
```diff
@@ -155,7 +156,8 @@ def _chat_attributes(self, is_selected: bool = False) -> Tuple[int]:
 
     def draw(self, current: int, chats: List[Dict[str, Any]]) -> None:
         self.win.erase()
-        self.win.vline(0, self.w - 1, curses.ACS_VLINE, self.h)
+        line = curses.ACS_VLINE  # type: ignore
+        self.win.vline(0, self.w - 1, line, self.h)
```
Comment (paul-nameless) [inline tg/views/__init__.py:160]: I don't like it, can it be disabled in mypy? It's forcing you to write some extra code which is stupid.

File: tg/views/__init__.py [hunk 3]
```diff
@@ -307,15 +311,15 @@ def draw(
 
         self._refresh()
 
-    def _msg_attributes(self, is_selected: bool) -> Tuple[int]:
+    def _msg_attributes(self, is_selected: bool) -> Tuple[int, ...]:
```
Comment (paul-nameless) [inline tg/views/__init__.py:314]: what ... means in type?

File: tg/main.py (hunk 1 above)
Comment (lightme16) [reply tg/main.py:8]: that was for missing import error
`../tg/main.py:8: error: Skipping analyzing 'telegram.client': found module but no type hints or library stubs`
ok, lets disable it with `--ignore-missing-imports` flag. 

Mypy doesn't check line lengths, that `noqa` was for my flake8, but let's remove that line for now

File: tg/views/__init__.py (hunk 2 above)
Comment (lightme16) [reply tg/views/__init__.py:160]: mypy is not ideal, here it seems that it contains outdated type stub for `curses` and it things that `ACS_VLINE` is not existing and we will get error.
The only way how to disabled it is to `type: ignore`.
Or you could disable all type check by stop using mypy completley :man_shrugging:

File: tg/views/__init__.py (hunk 3 above)
Comment (lightme16) [reply tg/views/__init__.py:314]: this annotation for tuple of variable size. plz see the [docs](https://mypy.readthedocs.io/en/stable/cheat_sheet_py3.html#built-in-types) mypy can't detect that tuple contains exactly 3 elements, that's why I am using it. 
Anyway, it is much better that wrong annotation `Tuple[int]:` which means tuple with one element inside

File: tg/views/__init__.py [hunk 4]
```diff
@@ -307,15 +313,15 @@ def draw(
...
         attrs = (
             get_color(cyan, -1),
             get_color(blue, -1),
             get_color(white, -1),
         )
 
         if is_selected:
-            return (attr | reverse for attr in attrs)
+            return tuple(attr | reverse for attr in attrs)
```
Comment (lightme16) [inline tg/views/__init__.py:324]: it was a generator, unit this PR :man_shrugging:

**Pull request merged**